src/pages/en/guides/deploy/github.md

```diff
@@ -25,45 +25,60 @@ You can deploy an Astro site to GitHub Pages by using [GitHub Actions](https://g
     name: Github Pages Astro CI
 
     on:
-      # Trigger the workflow every time you push to the `main` branch
-      # Using a different branch name? Replace `main` with your branch’s name
+     # Trigger the workflow every time you push to the `main` branch
       push:
-        branches: [main]
+        branches: [ main ]
       # Allows you to run this workflow manually from the Actions tab on GitHub.
       workflow_dispatch:
+      
+      # Allow this job to push changes to your repository
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
 
     jobs:
-      deploy:
-        runs-on: ubuntu-20.04
-
-        # Allow this job to push changes to your repository
-        permissions:
-          contents: write
-
+      build:
+        runs-on: ubuntu-latest
         steps:
-          - name: Check out your repository using git
-            uses: actions/checkout@v2
-
-          - name: Use Node.js 16
-            uses: actions/setup-node@v2
-            with:
-              node-version: 16
-
-          # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
-          - name: Install dependencies
-            run: npm ci
-
-          # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
-          - name: Build Astro
-            run: npm run build
+        - name: Check out your repository using git
+          uses: actions/checkout@v2
+
+        - name: Use Node.js 16
+          uses: actions/setup-node@v2
+          with:
+            node-version: '16'
+            cache: 'npm'
+
+        # Not using npm? Change `npm ci` to `yarn install` or `pnpm i`
+        - name: Install dependencies
+          run: npm ci
+
+        # Not using npm? Change `npm run build` to `yarn build` or `pnpm run build`
+        - name: Build Astro
+          run: npm run build --if-present
+
+        - name: Archive build output
+          # `./dist` is the default Astro build directory.
+          # If you changed that, update it here too.
+          run: "tar --dereference --directory dist/ -cvf artifact.tar ."
+
+        - name: Upload artifact
+          uses: actions/upload-artifact@v2
+          with:
+            name: github-pages
+            path: artifact.tar
 
+      deploy:
+        environment:
+          name: github-pages
+          url: ${{ steps.deployment.outputs.page_url }}
+        runs-on: ubuntu-latest
+        needs: build
+        steps:
           - name: Deploy to GitHub Pages
-            uses: peaceiris/actions-gh-pages@v3
-            with:
-              github_token: ${{ secrets.GITHUB_TOKEN }}
-              # `./dist` is the default Astro build directory.
-              # If you changed that, update it here too.
-              publish_dir: ./dist
+            id: deployment
+            uses: paper-spa/deploy-pages@main
```

Comment:
This is an org we use for staging and could break since it's from `@main`. Better to use the released actions from the actions org

```suggestion
            uses: actions/deploy-pages@v1
```